src/wayland-egldisplay.c

```diff
@@ -730,7 +730,7 @@ static EGLBoolean terminateDisplay(WlEglDisplay *display, EGLBoolean globalTeard
     wlEglDestroyAllSurfaces(display);
 
     wlEglDestroyFormatSet(&display->formatSet);
-    wlEglDestroyFeedback(&display->defaultFeedback);
+    wlEglDestroyFeedback(&display->defaultFeedback, globalTeardown);
```

Comment:
I actually think it might make more sense to move this call down into the `if (!globalTeardown || display->ownNativeDpy) {..}` block below this, just above the `if (display->wlDmaBuf) ` part. That would prevent us from having to change `wlEglDestroyFeedback` at all and also respect the `display->ownNativeDpy` checking we do. As-is I think we would defer `wlEglDestroyFeedback` but destroy everything else.

Reply:
I did it like that because wlEglDestroyFeedback also has a munmap call which isn't called this way but as library unload usually means application shutdown anyway I guess it's reasonable to ignore this case.

Reply:
Well right now since it isn't checked like you said it's possible for the `munmap` to happen twice if `terminateDisplay` is called multiple times for the deferred free workaround, so that's another problem fixed by this suggestion. If we move `wlEglDestroyFeedback` into the guarded `globalTeardown` section then we avoid this as well.